### ISSUE TITLE
pci_decode: prepare PCI device information for platform_drv

### ISSUE THREAD

**Issue by skalk** [genodelabs/genode#4495]
To support PCI devices within the current, architecture-independent platform driver, it would be good to separate the PCI bus scan, ACPI information and platform information detection from the platform driver logic.

Some more details: For instance on several ARM platforms we might have a PCI-x controller, which needs to be initialized dependent on the platform driver's resources, we do not have ACPI nor do we support MSIs currently. On the other hand, on x86-based PCs we do have an already initialized and working PCI/PCI-x system with ACPI information and MSI support in certain kernels. In the legacy x86-only platform driver we combined such information to e.g. decide to use PIC-only interrupts on certain old kernels, APIC legacy-interrups when a kernel does not support MSIs, and MSIs or MSI-x when the kernel and the PCI device suports it.
To sum it up: the information to collect and interpret are quite different, and the configuration of the PCI config for a specific PCI device function is dependent on it. To not "pollute" the generic platform driver too much with it, and keep its confirmability, the PCI device information preparation could be implemented separatedly.

This is one step to equip the generic platform driver with PCI device support.
